### ISSUE TITLE
`forge test --debug` crashes my entire computer

### ISSUE THREAD

**Issue by marktoda** [foundry-rs/foundry#2870]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (29ff0b0 2022-08-20T00:03:58.583580886Z)

### What command(s) is the bug in?

forge test --debug

### Operating System

Linux

### Describe the bug

Was using `forge test --debug` during the paradigm ctf and on `fun-reversing` halfway through the challenge the debugger crashed my computer lmao. I was able to capture some output by logging to file while my computer was crashing, but it doesn't seem particularly helpful. It's reproducible so let me know if anything else would help debug this. 

```
earlier... > RUST_BACKTRACE=1 RUST_LOG=forge,foundry_evm,ui forge test --debug testReversing > error.txt


> cat error.txt
2022-08-22T02:22:50.186313Z TRACE forge::test: executing test command
2022-08-22T02:22:50.190891Z TRACE forge::test: using filter filter=Filter { match: None, match-test: None, no-match-test: None, match-contract: None, no-match-contract: None, match-path: None, no-match-path: None, .. }
2022-08-22T02:22:50.190989Z TRACE forge::compile: start compiling project
Compiling 7 files with 0.8.15
Compiling 1 files with 0.7.6
Solc 0.7.6 finished in 856.48ms
Solc 0.8.15 finished in 944.67ms
2022-08-22T02:22:51.724008Z TRACE forge::compile: finished compiling after 1.533014234s
Compiler run successful
2022-08-22T02:22:51.724080Z TRACE foundry_evm::executor::inspector::cheatcodes::config: using resolved rpc endpoints rpc_endpoints=ResolvedRpcEndpoints { endpoints: {"paradigm": Ok("http://127.0.0.1:8545")} }
2022-08-22T02:22:51.802603Z TRACE forge::multi_runner: start executing all tests in contract contract="test/Reversing.t.sol:ReversingTest"
2022-08-22T02:22:51.802657Z  INFO contract{name=test/Reversing.t.sol:ReversingTest}: forge::runner: starting tests
2022-08-22T02:22:51.802664Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: forge::runner: Setting test contract setup=true
2022-08-22T02:22:51.802688Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor: deploying contract sender=0x00a329c0648769a73afac7f9381e08fb43dbea72
2022-08-22T02:22:51.802868Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor::backend: add persistent account account=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.802874Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor: deployed contract address=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.802880Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor: deploying contract sender=0x3fab184622dc19b6109349b94811493bf2a45362
2022-08-22T02:22:51.802896Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor::backend: add persistent account account=0x4e59b44847b379578588920ca78fbf26c0b4956c
2022-08-22T02:22:51.802898Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor: deployed contract address=0x4e59b44847b379578588920ca78fbf26c0b4956c
2022-08-22T02:22:51.802901Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: forge::runner: setting up
2022-08-22T02:22:51.802904Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor::backend: setting test account acc=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.802906Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor::backend: add persistent account account=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.802908Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor::backend: setting caller account acc=0x1804c8ab1f12e6bbf3894d4083f33e07309d1f38
2022-08-22T02:22:51.802919Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: foundry_evm::executor::backend: add persistent account account=0x1804c8ab1f12e6bbf3894d4083f33e07309d1f38
2022-08-22T02:22:51.802940Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}: forge::runner: successfully setUp test contract=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.802958Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: setting caller account acc=0x00a329c0648769a73afac7f9381e08fb43dbea72
2022-08-22T02:22:51.802961Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: add persistent account account=0x00a329c0648769a73afac7f9381e08fb43dbea72
2022-08-22T02:22:51.802964Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: setting test account acc=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.802966Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: remove persistent account account=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.802969Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: add persistent account account=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.803021Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: create fork
2022-08-22T02:22:51.803024Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::fork::multi: Creating new fork, url=http://127.0.0.1:8545, block=None
2022-08-22T02:22:51.803115Z TRACE foundry_evm::executor::fork::multi: created new forkId fork_id=ForkId("http://127.0.0.1:8545@latest")
2022-08-22T02:22:51.891157Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: select fork id=0
2022-08-22T02:22:51.891206Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::fork::multi: getting env config fork=ForkId("http://127.0.0.1:8545@latest")
2022-08-22T02:22:51.891347Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: recording fork init subroutine
2022-08-22T02:22:51.891379Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: cloning database data addr=0x00a329c0648769a73afac7f9381e08fb43dbea72
2022-08-22T02:22:51.891409Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: updating subroutine account data addr=0x00a329c0648769a73afac7f9381e08fb43dbea72
2022-08-22T02:22:51.891421Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: cloning database data addr=0x4e59b44847b379578588920ca78fbf26c0b4956c
2022-08-22T02:22:51.891432Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: cloning database data addr=0x1804c8ab1f12e6bbf3894d4083f33e07309d1f38
2022-08-22T02:22:51.891442Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: cloning database data addr=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.891454Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: updating subroutine account data addr=0xb4c79dab8f259c7aee6e5b2aa729821864227e84
2022-08-22T02:22:51.891463Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: cloning database data addr=0x7109709ecfa91a80626ff3989d68f67f5b1dd12d
2022-08-22T02:22:51.891475Z TRACE contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: foundry_evm::executor::backend: updating subroutine account data addr=0x7109709ecfa91a80626ff3989d68f67f5b1dd12d
2022-08-22T02:22:52.185442Z DEBUG contract{name=test/Reversing.t.sol:ReversingTest}:test{name=testReversing() should_fail=false}: forge::runner: duration=382.480802ms success=false gas=2959427
2022-08-22T02:22:52.185539Z  INFO contract{name=test/Reversing.t.sol:ReversingTest}: forge::runner: done. 0/1 successful duration=382.876004ms
2022-08-22T02:22:52.185586Z TRACE forge::multi_runner: executed all tests in contract contract="test/Reversing.t.sol:ReversingTest"
2022-08-22T02:22:52.227115Z TRACE forge::compile: start compiling project
Compiling 7 files with 0.8.16
Compiling 1 files with 0.7.6
Solc 0.7.6 finished in 853.22ms
Solc 0.8.16 finished in 2.21s
2022-08-22T02:22:54.955381Z TRACE forge::compile: finished compiling after 2.728264943s
Compiler run successful (with warnings)
warning[2072]: Warning: Unused local variable.
  --> src/rescue/Exploiter.sol:47:10:
   |
47 |         (address lpToken, ,,) = helper.masterchef().poolInfo(0);
   |          ^^^^^^^^^^^^^^^



2022-08-22T02:22:55.165052Z TRACE foundry_evm::executor::inspector::cheatcodes::config: using resolved rpc endpoints rpc_endpoints=ResolvedRpcEndpoints { endpoints: {"paradigm": Ok("http://127.0.0.1:8545")} }
2022-08-22T02:22:55.165098Z TRACE foundry_evm::executor: deploying contract sender=0x3fab184622dc19b6109349b94811493bf2a45362
2022-08-22T02:22:55.165139Z TRACE foundry_evm::executor::backend: add persistent account account=0x4e59b44847b379578588920ca78fbf26c0b4956c
2022-08-22T02:22:55.165142Z TRACE foundry_evm::executor: deployed contract address=0x4e59b44847b379578588920ca78fbf26c0b4956c
2022-08-22T02:22:55.165146Z TRACE foundry_evm::executor: deploying contract sender=0x1804c8ab1f12e6bbf3894d4083f33e07309d1f38
2022-08-22T02:22:55.165242Z TRACE foundry_evm::executor::backend: add persistent account account=0x5b73c5498c1e3b4dba84de0f1833c4a029d90519
2022-08-22T02:22:55.165245Z TRACE foundry_evm::executor: deployed contract address=0x5b73c5498c1e3b4dba84de0f1833c4a029d90519
2022-08-22T02:22:55.165248Z TRACE foundry_evm::executor::backend: setting test account acc=0x5b73c5498c1e3b4dba84de0f1833c4a029d90519
2022-08-22T02:22:55.165250Z TRACE foundry_evm::executor::backend: add persistent account account=0x5b73c5498c1e3b4dba84de0f1833c4a029d90519
2022-08-22T02:22:55.165252Z TRACE foundry_evm::executor::backend: setting caller account acc=0x00a329c0648769a73afac7f9381e08fb43dbea72
2022-08-22T02:22:55.165254Z TRACE foundry_evm::executor::backend: add persistent account account=0x00a329c0648769a73afac7f9381e08fb43dbea72
2022-08-22T02:22:55.165364Z TRACE foundry_evm::executor::backend::fuzz: fuzz: create fork
2022-08-22T02:22:55.165369Z TRACE foundry_evm::executor::backend: create fork
2022-08-22T02:22:55.165372Z TRACE foundry_evm::executor::fork::multi: Creating new fork, url=http://127.0.0.1:8545, block=None
2022-08-22T02:22:55.165424Z TRACE foundry_evm::executor::fork::multi: created new forkId fork_id=ForkId("http://127.0.0.1:8545@latest")
2022-08-22T02:22:55.166709Z TRACE foundry_evm::executor::backend::fuzz: fuzz: select fork id=0
2022-08-22T02:22:55.166717Z TRACE foundry_evm::executor::backend: select fork id=0
2022-08-22T02:22:55.166720Z TRACE foundry_evm::executor::fork::multi: getting env config fork=ForkId("http://127.0.0.1:8545@latest")
2022-08-22T02:22:55.166759Z TRACE foundry_evm::executor::backend: recording fork init subroutine
2022-08-22T02:22:55.166768Z TRACE foundry_evm::executor::backend: cloning database data addr=0x7109709ecfa91a80626ff3989d68f67f5b1dd12d
2022-08-22T02:22:55.166774Z TRACE foundry_evm::executor::backend: updating subroutine account data addr=0x7109709ecfa91a80626ff3989d68f67f5b1dd12d
2022-08-22T02:22:55.166778Z TRACE foundry_evm::executor::backend: cloning database data addr=0x4e59b44847b379578588920ca78fbf26c0b4956c
2022-08-22T02:22:55.166781Z TRACE foundry_evm::executor::backend: cloning database data addr=0x5b73c5498c1e3b4dba84de0f1833c4a029d90519
2022-08-22T02:22:55.166785Z TRACE foundry_evm::executor::backend: updating subroutine account data addr=0x5b73c5498c1e3b4dba84de0f1833c4a029d90519
2022-08-22T02:22:55.166793Z TRACE foundry_evm::executor::backend: cloning database data addr=0x00a329c0648769a73afac7f9381e08fb43dbea72
2022-08-22T02:22:55.166796Z TRACE foundry_evm::executor::backend: updating subroutine account data addr=0x00a329c0648769a73afac7f9381e08fb43dbea72
2022-08-22T02:22:55.166799Z TRACE foundry_evm::executor::backend: cloning database data addr=0x1804c8ab1f12e6bbf3894d4083f33e07309d1f38
```

**Comment by plotchy** [comment]
Same experience on this challenge. Seems to be related to the amount of memory expansion the challenge program creates alongside the amount of opcodes needed to step through. 

I'm unsure of the root cause, we were thinking that memory state is being stored per opcode so long traces blow up to GB+?

cc @brockelmore

**Comment by mattsse** [comment]
this should be resolved now with #2877

please reopen if you see this again
